### PR TITLE
Ident scanning, single quote strings

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This pull request changes the scanner to treat double quoted strings as identifiers and single quoted strings as strings. The scanner can also chain multiple quoted or bare idents together if they are joined by a dot.

Identifiers:

```
"db"."rp".cpu
"db"..cpu
"cpu"
cpu
```

Strings:

```
'hello world'
'two\nlines'
'\'i ♡ quotes\''
```

### CODE REVIEW

**Comment by dgnorton** [comment]
There are ~14 places in parser.go that still use `tok != IDENT && tok != STRING` where it's looking for an identifier or string.  I've been meaning to convert them to use `parseIdentifier` or `parseString` as needed.  That probably needs to be done as part of this PR.

**Comment by benbjohnson** [comment]
@dgnorton Fixed: 73bc821

**Comment by dgnorton** [comment]
lgtm

File: influxql/parser_test.go [hunk 1]
```diff
@@ -338,7 +337,7 @@ func TestParser_ParseStatement(t *testing.T) {
 
 		// CREATE USER statement
 		{
-			s: `CREATE USER testuser WITH PASSWORD pwd1337`,
+			s: `CREATE USER testuser WITH PASSWORD 'pwd1337'`,
```
Comment (otoolep) [inline influxql/parser_test.go]: So we have to quote passwords now? I thought the parser would know to interpret whatever follows `PASSWORD` as a string. Am I mistaken?

Comment (benbjohnson) [inline influxql/parser_test.go]: Users will need single quotes. It's a string. It's better than the SQL version which requires people to wrap the password in a call: `PASSWORD('xxx')`

http://dev.mysql.com/doc/refman/5.0/en/set-password.html

Comment (otoolep) [inline influxql/parser_test.go]: Surely the SQL example we should compare against is `CREATE USER` which doesn't require wrapping the password in call?

http://dev.mysql.com/doc/refman/5.0/en/create-user.html

However, my experiments show that MySQL also requires the password to be quoted (something I didn't realise) so we're as good as them. :-)

**Pull request merged**